### PR TITLE
scripts/test.sh: add batching to markdownlint

### LINKED ISSUE #9468: Git hook issue: "markdownlint: Argument list too long"
I started having an issue where I'm unable to commit on MacOS using bash as shell (zsh works fine, but old habits remain).

Running `npm test` or trying to commit gives me a message like:

```
bash-3.2$ npm test

> test
> bash scripts/test.sh

scripts/test.sh: line 48: /Users/chanibal/Projekty/Prywatne/tldr/node_modules/.bin/markdownlint: Argument list too long
```

A quick fix is to run it from zsh:
```
bash-3.2$ zsh
chanibal@MacBook-Pro-Krzysztof tldr % npm test

> test
> bash scripts/test.sh

Skipping black check, command not available.
Skipping flake8 check, command not available.
Test ran successfully!
```

Probably caused by, well, a lot of arguments given to markdownlint. At the time of writing, 7703 filenames or 196698 bytes, no idea why this is the limit, but it stopped working for me. Bash's `getconf ARG_MAX` is at 262144, so there is still some wiggle room. Needs batching.

### PR DESCRIPTION
**Pull request by chanibal**
This fixes #9468 by adding batching to call of markdownlint.

### CODE REVIEW

**Review by navarroaxel — APPROVED** [review]

**Review by kbdharun — APPROVED** [review]

**Comment by kbdharun** [comment]
Can this be merged? or should we request more reviews @navarroaxel?

**Comment by EmilySeville7cfg** [comment]
> Can this be merged? or should we request more reviews @navarroaxel?

Everything can be merged when there are two approvals according to our current maintainers guide. Now there are even three. 😸

**Pull request merged**
